pkg/solana/chainwriter/ccip_example_config.go

```diff
@@ -0,0 +1,333 @@
+				// In this case, the user configured multiple seeds. These will be used in conjunction
+				// with the public key to generate one or multiple PDA accounts.
+				Seeds: []Lookup{
+					AccountLookup{Location: "Message.Receiver"},
+					AccountLookup{Location: "Message.Header.DestChainSelector"},
+				},
+			},
+			// Account constant
+			AccountConstant{
+				Name:       "CPISigner",
+				Address:    cpiSignerAddress,
+				IsSigner:   true,
+				IsWritable: false,
+			},
+			// Account constant
+			AccountConstant{
+				Name:       "SystemProgram",
+				Address:    systemProgramAddress,
+				IsSigner:   true,
+				IsWritable: false,
+			},
+			// Account constant
+			AccountConstant{
+				Name:       "ComputeBudgetProgram",
+				Address:    computeBudgetProgramAddress,
+				IsSigner:   true,
+				IsWritable: false,
+			},
+			// Account constant
+			AccountConstant{
+				Name:       "SysvarProgram",
+				Address:    sysvarProgramAddress,
+				IsSigner:   true,
+				IsWritable: false,
+			},
+		},
+		// TBD where this will be in the report
+		// This will be appended to every error message
+		DebugIDLocation: "Message.MessageID",
+	}
+
+	commitConfig := MethodConfig{
+		FromAddress:        userAddress,
+		InputModifications: nil,
+		ChainSpecificName:  "commit",
+		LookupTables: LookupTables{
+			StaticLookupTables: []string{
+				commonAddressesLookupTable,
+				routerLookupTable,
+			},
+		},
+		Accounts: []Lookup{
+			// Account constant
+			AccountConstant{
+				Name:       "RouterProgram",
+				Address:    routerProgramAddress,
+				IsSigner:   false,
+				IsWritable: false,
+			},
+			// Account constant
+			AccountConstant{
+				Name:       "RouterAccountConfig",
+				Address:    routerAccountConfigAddress,
+				IsSigner:   false,
+				IsWritable: false,
+			},
+			// PDA lookup to get the Router Report Accounts.
+			PDALookups{
+				Name: "RouterReportAccount",
+				// The public key is a constant Router address.
+				PublicKey: AccountConstant{
+					Address:    routerProgramAddress,
+					IsSigner:   false,
+					IsWritable: false,
+				},
+				Seeds: []Lookup{
+					AccountLookup{
+						// The seed is the merkle root of the report, as passed into the input params.
+						Location: "args.MerkleRoots",
+					},
+				},
+				IsSigner:   false,
+				IsWritable: false,
+			},
+			// Account constant
+			AccountConstant{
+				Name:       "SystemProgram",
+				Address:    systemProgramAddress,
+				IsSigner:   true,
+				IsWritable: false,
+			},
+			// Account constant
+			AccountConstant{
+				Name:       "ComputeBudgetProgram",
+				Address:    computeBudgetProgramAddress,
+				IsSigner:   true,
+				IsWritable: false,
+			},
+			// Account constant
+			AccountConstant{
+				Name:       "SysvarProgram",
+				Address:    sysvarProgramAddress,
+				IsSigner:   true,
+				IsWritable: false,
+			},
+		},
+		DebugIDLocation: "",
+	}
+
+	chainWriterConfig := ChainWriterConfig{
+		Programs: map[string]ProgramConfig{
+			"ccip-router": {
```

Comment:
To make this example accurate, I think this program name needs to be a public key

Reply:
The Program Config's key can be any string, it doesn't have to be a PublicKey. The program address is passed through the `toAddress` param in `SubmitTransaction`

Reply:
Ah ok then I think there's a change needed in `SubmitTransaction`. I'll add a comment at the spot. I ran into this issue when writing the unit tests which is why I changed the program config key to an address.